### PR TITLE
Timeout on `GPGraph` construction in `GreedyPauliSimp`

### PR DESCRIPTION
**Pull request by sjdilkes**
`GPGraph` construction and `GPGraph::get_sequence` can run significantly longer than the given timeout. This PR updates `greedy_pauli_graph_synthesis_flag` to check timeout during the execution of these methods and to finish early if the timeout is met.

# Related issues

Please mention any github issues addressed by this PR.

# Checklist

- [ ] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

File: tket/include/tket/Transformations/GreedyPauliOptimisation.hpp [hunk 1]
```diff
@@ -525,8 +525,11 @@ typedef boost::adj_list_vertex_property_map<
  */
 class GPGraph {
  public:
-  /** Construct an GPGraph from a circuit */
-  GPGraph(const Circuit& circ);
+  /** Construct an GPGraph from the circuit size */
```
Comment (cqc-alec) [inline tket/include/tket/Transformations/GreedyPauliOptimisation.hpp:528]:
```suggestion
  /** Construct an GPGraph from the qubits and bits */
```
?

File: tket/include/tket/Transformations/GreedyPauliOptimisation.hpp [hunk 2]
```diff
@@ -525,8 +525,11 @@ typedef boost::adj_list_vertex_property_map<
...
+  GPGraph(qubit_vector_t qubits, bit_vector_t bits);
+
+  GPGraph get_graph(
```
Comment (cqc-alec) [inline tket/include/tket/Transformations/GreedyPauliOptimisation.hpp:531]: Can you document what this function does?

Comment (sjdilkes) [reply tket/include/tket/Transformations/GreedyPauliOptimisation.hpp:531]: Ah, that method shouldn't even be there & is a leftover from an alternative way I considered doing this. I've removed it!

File: tket/test/src/test_GreedyPauli.cpp [hunk 3]
```diff
@@ -764,70 +764,72 @@ SCENARIO("Test GreedyPauliSimp pass construction") {
...
+          {0, 5, 1, 4, 3, 2});
+    }
     Circuit d(circ);
     REQUIRE(!Transforms::greedy_pauli_optimisation(
-                 0.7, 0.3, 500, 500, 0, true, 0, 10)
+                 0.7, 0.3, 500, 500, 0, true, 1, 10)
                  .apply(d));
     REQUIRE(Transforms::greedy_pauli_optimisation(
-                0.7, 0.3, 500, 500, 0, true, 10, 10)
+                0.7, 0.3, 500, 500, 0, true, 10000, 10)
```
[the remark this answers is not in the document]
Comment (sjdilkes) [reply tket/test/src/test_GreedyPauli.cpp]: done! I've changed the loop size to make the circuit a 1/5 of the size

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**